File: backend/app/models/notification.py
from uuid import UUID
from pydantic import BaseModel
from datetime import datetime
from typing import Dict, Any, Optional


class BaseNotification(BaseModel):
    client_id: Optional[UUID]
    technician_id: Optional[UUID]
    title: str
    message: str


class NotificationInDB(BaseNotification):
    id: UUID
    is_read: bool
    created_at: datetime


class NotificationCreate(BaseNotification):

    class Config:
        json_schema_extra: Dict[str, Any] = {
            "example": {
                "technician_id": "123e4567-e89b-12d3-a456-426655440000",
                "client_id": None,
                "title": "Booking Request",
                "message": "You have a new booking request about plumbing from Tumelo Modise in Soshanguve Block L",
            }
        }


class NotificationResponse(NotificationInDB):

    class Config:
        json_schema_extra: Dict[str, Any] = {
            "example": {
                "id": "123e4567-e89b-12d3-a456-426655440000",
                "technician_id": "123e4567-e89b-12d3-a456-426655440000",
                "client_id": None,
                "title": "Booking Request",
                "message": "You have a new booking request about plumbing from Tumelo Modise in Soshanguve Block L",
                "is_read": False,
                "created_at": "2025-06-01T00:00:00",
            }
        }

File: frontend/src/lib/types/notification.ts
import { z } from "zod";

const BaseNotificationSchema = z.object({
    client_id: z.string().nullable(),
    technician_id: z.string().nullable(),
    title: z.string(),
    message: z.string()
});

export const NotificationCreateSchema = BaseNotificationSchema.extend({});

export const NotificationResponseSchema = BaseNotificationSchema.extend({
    id: z.string().uuid(),
    is_read: z.boolean(),
    created_at: z.string().datetime(),
});

export type NotificationCreate = z.infer<typeof NotificationCreateSchema>;
export type NotificationResponse = z.infer<typeof NotificationResponseSchema>;
